src/main/java/dk/kb/datahandler/kaltura/KalturaDeltaUploadJob.java

```diff
@@ -0,0 +1,304 @@
+                   long mTime= (Long) doc.getFieldValue("internal_storage_mTime");
+                   
+                   mTimeFromCurrent=mTime; //update mTime for next call                   
+                                 
+                 try {                                    
+                   //upload stream
+                   MediaType mediaType = KalturaUtil.getMediaType(resourceDescription);
+                   int flavourParamId = KalturaUtil.getFlavourParamId(mediaType);
+                   String path =KalturaUtil.generateStreamPath(fileId, originatesFrom, resourceDescription);
+                   log.info("validating stream='{}' with title='{}'",path,title);
+                 
+                   String fileError= hasStreamFileError(path, minimumFileSizeInBytes);
+                   if (fileError != null) {
+                       log.warn("File does not exist='{}' or size in bytes less than '{}'. Error='{}'. Id='{}'. Skipping upload", path, minimumFileSizeInBytes,fileError,id);     
+                       updateKalturaIdForRecord(storageClient, fileId, fileError);
+                       continue;                       
+                   }
+                   
+                   // Check file not already in kaltura. 
+                   String kalturaInternalId=getInternalIdKaltura(fileId);
+                   if (kalturaInternalId != null) {
+                       log.warn("Stream allready found in kaltura. FileId='{}'. Setting this kalturaId for record.",kalturaInternalId  );
+                       updateKalturaIdForRecord(storageClient, fileId, kalturaInternalId);                       
+                       continue; //Do not upload stream
+                   }
+                   
+                   try {
+                     String kalturaId=uploadStream(title, fileId, description, path, uploadTagForKaltura,mediaType,flavourParamId);
+                     log.info("Upload stream='{}' and got kalturaId='{}'",path,kalturaId);
+                     numberStreamsUploaded++; //Success count
+                     //update storage record with kalturaId                     
+                     updateKalturaIdForRecord(storageClient, fileId, kalturaId);
+                     log.info("Updated kaltura mapping in storage for fileId:"+fileId);  
+                     continue; //Not required since nothing else happens below. But for consistency.
+                   }
+                   catch(Exception e) {  //Stop delta job
+                       log.error("Failed uploading stream to kaltura with fileId='{}', path='{}', title='{}', error='{}'",fileId,path,title,e.getMessage());
+                       updateKalturaIdForRecord(storageClient, fileId, StreamErrorTypeDto.API.getValue()); //Mark as API error
+                       throw new InternalServiceException("Failed uploading stream to kaltura with fileId="+fileId);
+                   }                                    
```

Comment:
This is quite a long method, is it possible to extract parts of it to individual methods?

Including the outer try-statement the method runs from line 107 to 152

Reply:
Line 107 was 152 extracted, reducing 1 level nested code.
Had to change a few things also.
continue -> return
Had to add the counter variable and change from primitive to object, so it will be updated visible for the outside code.

Code is still not pretty but hard to refactor further  without a rewrite. Problem is if splitting into more methods we need to know if we have to continue from return variable also.

Reply:
I like this better!